### ISSUE TITLE
Following the "Use lambda-case" suggestion can cause compile errors

### ISSUE THREAD

**Issue by tswelsh** [ndmitchell/hlint#457]
Given the file:

```
module LambdaCase where

f = \t -> case t of
  "a" -> "a"
  "b" -> "c"
```

`hlint` 2.1 suggests using lambda case (along with warning about the redundant lambda - this was distilled from a case where the redundant lambda warning wasn't present).  Naively following the suggestion results in a compile error as the `LambdaCase` extension isn't enabled.  While I know how to fix it it is mildly annoying, and could be even more so if you are in the context of automatically applying lots of suggestions to a file.  Options to improve it:

1. Don't suggest lambda case at all if the extension is not enabled.
1. Add a note to the hint along the lines of "You may need to enable the LambdaCase extension when applying this suggestion", maybe with an example of the extension line to add.
1. Add the same note but only when we can detect that `LambdaCase` isn't enabled.
1. Automatically add the LANGUAGE pragma to the top of the file if it isn't in effect.

As far as I can see hlint works on a single file at a time so detecting if an extension is actually in effect could give a false negative if the extension is defined in the cabal file.  If that's true then I suppose option 2 would be the least invasive.

**Comment by ndmitchell** [comment]
Thanks for the suggestion, a great idea! Your diagnosis is correct, including that 2 is the easiest option, but we can get some of 3 as well. I've modified it so the message is now:

```
Sample.hs:3:5: Suggestion: Use lambda-case
Found:
  \ t ->
    case t of
        "a" -> "a"
        "b" -> "c"
Why not:
  \case
      "a" -> "a"
      "b" -> "c"
Note: may require `{-# LANGUAGE LambdaCase #-}` adding to the top of the file
```

In addition, if you already have `LambdaCase` turned on in this file then I don't add the note.
